apps/api/v2/src/modules/organizations/repositories/organization-users.repository.ts

```diff
@@ -0,0 +1,73 @@
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import { Injectable } from "@nestjs/common";
+
+import { UpdateOrganizationUserInput_2024_06_18 } from "@calcom/platform-types";
+
+const userSelect = {
+  id: true,
+  username: true,
+  name: true,
+  email: true,
+  emailVerified: true,
+  bio: true,
+  avatarUrl: true,
+  timeZone: true,
+  weekStart: true,
+  appTheme: true,
+  theme: true,
+  defaultScheduleId: true,
+  locale: true,
+  timeFormat: true,
+  hideBranding: true,
+  brandColor: true,
+  darkBrandColor: true,
+  allowDynamicBooking: true,
+  createdDate: true,
+  verified: true,
+  invitedTo: true,
+  role: true,
+};
+
+@Injectable()
+export class OrganizationUsersRepository {
+  constructor(private readonly dbRead: PrismaReadService) {}
+
+  async getOrganizationUsers(organizationId: number, emailArray?: string[]) {
+    const usersQuery = await this.dbRead.prisma.user.findMany({
+      where: {
+        teams: {
+          some: {
```

Comment:
Can't we just check "User.organizationId" instead of checking "some: teams" ?

Reply:
I think we should do profile organizationId instead, as we discussed on the other PR